solver/src/settlement/settlement_encoder.rs

```diff
@@ -68,6 +68,17 @@ impl SettlementEncoder {
         result
     }
 
+    // Returns a copy of self without any liquidity provision interaction.
+    pub fn without_onchain_liquidity(&self) -> Self {
```

Comment:
Really cool PR! great start!

Maybe the next step is to write a fn: `without_onchain_stablecoin_liquidity` where we just replace the onchain stable coin liquidity but leave all other interactions, because the current solution might not work on WETH->USDC->DAI trading pairs if the buffer is not sufficient on WETH->USDC.

Reply:
It can be even better. We can also use buffers for one order even if the order is not totally covered by them since that will save us uniswap fees.